packages/components/src/tooltip/test/index.tsx

```diff
@@ -0,0 +1,276 @@
+
+		expect( button ).toBeVisible();
+		expect( button ).toHaveAttribute( 'aria-disabled' );
+
+		await user.hover( button );
+
+		expect(
+			await screen.findByRole( 'tooltip', { name: /tooltip text/i } )
+		).toBeVisible();
+
+		await cleanupTooltip( user );
+	} );
+
+	it( 'should not show tooltip if the mouse leaves the tooltip anchor before set delay', async () => {
+		const user = userEvent.setup();
+		const onMouseEnterMock = jest.fn();
+		const onMouseLeaveMock = jest.fn();
+		const MOUSE_LEAVE_DELAY = TOOLTIP_DELAY - 200;
+
+		render(
+			<>
+				<Tooltip { ...props }>
+					<Button
+						onMouseEnter={ onMouseEnterMock }
+						onMouseLeave={ onMouseLeaveMock }
+					>
+						Button 1
+					</Button>
+				</Tooltip>
+				<Button>Button 2</Button>
+			</>
+		);
+
+		await user.hover(
+			screen.getByRole( 'button', {
+				name: 'Button 1',
+			} )
+		);
+
+		// Tooltip hasn't appeared yet
+		expect(
+			screen.queryByRole( 'tooltip', { name: /tooltip text/i } )
+		).not.toBeInTheDocument();
+		expect( onMouseEnterMock ).toHaveBeenCalledTimes( 1 );
+
+		// Advance time by MOUSE_LEAVE_DELAY time
+		await new Promise( ( resolve ) =>
+			setTimeout( resolve, MOUSE_LEAVE_DELAY )
+		);
+
+		expect(
+			screen.queryByRole( 'tooltip', { name: /tooltip text/i } )
+		).not.toBeInTheDocument();
+
+		// Hover the other button, meaning that the mouse will leave the tooltip anchor
+		await user.hover(
+			screen.getByRole( 'button', {
+				name: 'Button 2',
+			} )
+		);
+
+		// Tooltip still hasn't appeared yet
+		expect(
+			screen.queryByRole( 'tooltip', { name: /tooltip text/i } )
+		).not.toBeInTheDocument();
+		expect( onMouseEnterMock ).toHaveBeenCalledTimes( 1 );
+		expect( onMouseLeaveMock ).toHaveBeenCalledTimes( 1 );
+
+		// Advance time again, so that we reach the full TOOLTIP_DELAY time
+		await new Promise( ( resolve ) =>
+			setTimeout( resolve, TOOLTIP_DELAY )
+		);
+
+		// Tooltip won't show, since the mouse has left the tooltip anchor
+		expect(
+			screen.queryByRole( 'tooltip', { name: /tooltip text/i } )
+		).not.toBeInTheDocument();
+
+		await cleanupTooltip( user );
+	} );
+
+	it( 'should render the shortcut display text when a string is passed as the shortcut', async () => {
+		const user = userEvent.setup();
+
+		render( <Tooltip { ...props } shortcut="shortcut text" /> );
+
+		await user.hover( screen.getByRole( 'button', { name: /Button/i } ) );
+
+		expect( await screen.findByText( 'shortcut text' ) ).toBeVisible();
+
+		await cleanupTooltip( user );
+	} );
+
+	it( 'should render the keyboard shortcut display text and aria-label when an object is passed as the shortcut', async () => {
+		const user = userEvent.setup();
+
+		render(
+			<Tooltip
+				{ ...props }
+				shortcut={ {
+					display: '⇧⌘,',
+					ariaLabel: shortcutAriaLabel.primaryShift( ',' ),
+				} }
+			/>
+		);
+
+		await user.hover( screen.getByRole( 'button', { name: /Button/i } ) );
+
+		await waitFor( () =>
```

Comment:
I tried to replace all `getBy` queries with `findBy` and therefore remove `waitFor()` but it did not work in this case